diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4d9cb06..fe815df 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,7 +79,10 @@ function(cst_bundle_libraries output_target)
   endforeach()
   foreach(dependency IN LISTS all_dependencies)
   get_target_property(type ${dependency} TYPE)
-  if(${type} STREQUAL "STATIC_LIBRARY")
+  get_target_property(is_imported ${dependency} IMPORTED)
+  if(is_imported)
+    list(APPEND all_imported_deps ${dependency})
+  elseif(${type} STREQUAL "STATIC_LIBRARY")
     list(APPEND all_objects $<TARGET_OBJECTS:${dependency}>)
   elseif(${type} STREQUAL "OBJECT_LIBRARY")
     list(APPEND all_objects $<TARGET_OBJECTS:${dependency}>)
@@ -93,6 +96,9 @@ function(cst_bundle_libraries output_target)
     add_library(${output_target} ${all_objects})
   endif()
   add_dependencies(${output_target} ${ARG_UNPARSED_ARGUMENTS})
+  if(all_imported_deps)
+    target_link_libraries(${output_target} PUBLIC ${all_imported_deps})
+  endif()
 endfunction()
 
 
@@ -323,17 +329,7 @@ if(SUPPORT_VULKAN_BACKEND)
     set(WGVK_BUILD_WGSL_SUPPORT ${SUPPORT_WGSL_PARSER})
     set(WGVK_BUILD_GLSL_SUPPORT ${SUPPORT_GLSL_PARSER})
     
-    if(NOT EXISTS "${CMAKE_CURRENT_LIST_DIR}/../WGVK")
-        FetchContent_Declare(
-            wgvk
-            GIT_REPOSITORY "https://github.com/manuel5975p/WGVK"
-            GIT_TAG master
-            GIT_SHALLOW True
-        )
-        FetchContent_MakeAvailable(wgvk)
-    else()
-        add_subdirectory("${CMAKE_CURRENT_LIST_DIR}/../WGVK" ${CMAKE_CURRENT_BINARY_DIR}/wgvk-build)
-    endif()
+    find_package(wgvk CONFIG REQUIRED)
 endif()
 if(APPLE)
     set_source_files_properties("src/InitWindow.c"
@@ -451,7 +447,7 @@ else()
     endif()
     if(SUPPORT_VULKAN_BACKEND)
         target_include_directories(${raygpu_core_library_name} PRIVATE "amalgamation/vulkan_headers/include")
-        list(APPEND raygpu_linked_deps wgvk)
+        list(APPEND raygpu_linked_deps wgvk::wgvk)
     else()
         list(APPEND raygpu_linked_deps webgpu_dawn)
     endif()
@@ -472,7 +468,7 @@ endif()
 target_link_libraries(${raygpu_core_library_name} PUBLIC ${raygpu_linked_deps})
 
 if(NOT SUPPORT_WGPU_BACKEND) 
-    target_link_libraries(${raygpu_core_library_name} PUBLIC wgvk)
+    target_link_libraries(${raygpu_core_library_name} PUBLIC wgvk::wgvk)
 endif()
 
 # Abseil's symbolize library requires linking dbghelp on windows, needing an
